code/game/objects/items/stacks/stack.dm

```diff
@@ -296,6 +296,23 @@ Also change the icon to reflect the amount of sheets, if possible.*/
 		if(!oldsrc)
 			break
 
+/obj/item/stack/clicked(mob/user, list/mods)
+	if(mods["alt"])
+		var/desired = tgui_input_number(user, "How much would you like to split off from this stack?", "How much?", 1, amount-1, 1)
+		if(!desired)
+			return
+		var/obj/item/stack/newstack = new src.type(user, desired)
```

Comment:
Feels like this should once again check that desired is indeed less than amount (what if you altclick it several times in a row and only then input numbers, for example?). Something like `if(!use(desired)){return}` before making new stack instead of `use(desired)` after.